src/ngx_http_lua_initworkerby.c

```diff
@@ -40,10 +40,30 @@ ngx_http_lua_init_worker(ngx_cycle_t *cycle)
 
     lmcf = ngx_http_cycle_get_module_main_conf(cycle, ngx_http_lua_module);
 
-    if (lmcf == NULL
-        || lmcf->init_worker_handler == NULL
-        || lmcf->lua == NULL)
-    {
+    if (lmcf == NULL || lmcf->lua == NULL) {
+        return NGX_OK;
+    }
+
+    /* lmcf != NULL && lmcf->lua != NULL */
+
+    /* disable init_worker_by_lua* and destroy lua VM in cache processes */
+    if (ngx_process == NGX_PROCESS_HELPER) {
+#if defined(HAVE_PRIVILEGED_PROCESS_PATCH) && !NGX_WIN32
+        if (!ngx_is_privileged_agent) {
```

Comment:
Seems like we can just use `&&` instead of `if` here. This way we can also save the `#if/#endif` pair below.